app/javascript/components/widgets/buses/panel.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useRef, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import mapboxgl from '!mapbox-gl';
+import PropTypes from 'prop-types';
+import geoJson from './chicago-parks.json';
+import MapStyle from './globalMapStyle';
+import vehiclepositions from './vehiclepositions.json';
+
+mapboxgl.accessToken = process.env.MAPBOX_TOKEN;
+
+const MapDimensions = styled.div`
+  width: 1000px;
+  height: 750px;
+`;
+
+const Marker = ({ feature, map }) => {
+  const markerref = useRef(null);
+
+  useEffect(() => {
+    const marker = new mapboxgl.Marker(markerref)
+      .setLngLat([
+        feature.vehicle.position.longitude,
+        feature.vehicle.position.latitude,
+      ])
+
+      .addTo(map);
+    return () => marker.remove();
+  });
+
+  return <div ref={markerref} className="marker" />;
```

Comment:
It looks like we're using `useEffect` to call the mapgl library. How does the ref relatet to the options object the `Marker` method accepts? Why do we connect that to a div? Do we need react here at all?